### ISSUE TITLE
Check if getData() had data

### ISSUE THREAD

**Issue by jonaseberledmind** [aimeos/ai-admin-jqadm#6]
in admin/jqadm/themes/dashboard.js and in some other places, the line 
```
if( data.meta && data.meta.prefix ) {
```
does not check if "data" might be undefined.

In our case when opening the dashboard we get the JS error "TypeError: data is undefined" at this point. 

The underlying problem seems to be in our case that there is a CORS preflight (although it shouldn't be IMHO because the request is to the same server that delivered the HTML) that does not respond with the expected JSON but (correctly) with a Status 204 and CORS-headers. As I see it atm, the XHR does not seem to handle CORS transparently but sets data = undefined.

As a first step it would be nice though to see that there was a transmission error.

**Comment by aimeos** [comment]
Would be a good thing to handle CORS responses correctly. There's not much documentation of how to do this correctly when using JQuery but maybe only `withCredentials: true` one is missing:
```
$.ajax({
   url: a_cross_domain_url,
   xhrFields: {
      withCredentials: true
   }
});
```
We can check for data too and throw an error if it's not available.

**Comment by aimeos** [comment]
@jonaseberledmind Can you check if adding `withCredentials: true` to dashboard.js helps?
https://github.com/aimeos/ai-admin-jqadm/blob/master/admin/jqadm/themes/dashboard.js#L42

**Comment by jonaseberledmind** [comment]
Sorry, that does not seem to help.

**Comment by aimeos** [comment]
Do you have any other idea how we can fix that? If we only check for `data` that doesn't mean that it will work for you

**Comment by jonaseberledmind** [comment]
I fixed it. It was a problem on my side. 
When I looked at the body inside the AJAX-request (method: OPTIONS) I saw an error there:
Class "\Aimeos\MShop\Product\Manager\Planeaffected\Plane\Type\Standard" not available ... ;)

I fixed that error and now the AJAX-requests are working again. 


**If you want to read the things I checked before, go on here:**

I don't understand at the moment what's going on there.
I wouldn't expect any CORS-preflight under these circumstances.

I disabled HTTP-Auth.

I set the "Access-Control-Allow-*"-headers to "allow anything", even looking closely at the incoming headers and explicitly allowing them.

I mainly wondered why the AJAX-request was not executed after the successful preflight.

Then I found this:

ext/ai-admin-jqadm/admin/jqadm/themes/admin.js :
```
Aimeos.options = $.ajax($(".aimeos").data("url"), {
	"method": "OPTIONS",
	"dataType": "json"
});

```

It seems aimeos is doing an OPTIONS-request deliberately, not as an actual CORS-preflight. 

I think we are not really having a CORS-problem here but just some processing that expects data where no data/non-JSON-data is returned:


ext/ai-admin-jqadm/admin/jqadm/themes/catalog.js:
```
Aimeos.options.done(function(data) {
	var rootId = $(".aimeos .item-catalog").data("rootid");
	if(!rootId || !data['meta'] || !data['meta']['resources'] || !data['meta']['resources']['catalog']) {
		return;
	}
```
(fails due to data being undefined)


The problem could also be that jQuery.ajax() with dataType:'json' might not handle the Content-Type "application/vnd.api+json" correctly which is returned by the JsonApi client?

I don't know...
